server/embed/etcd.go

```diff
@@ -249,6 +258,15 @@ func StartEtcd(inCfg *Config) (e *Etcd, err error) {
 		return e, err
 	}
 
+	if e.cfg.ExperimentalEnableDistributedTracing {
+		tctx := context.Background()
+		tracingExporter, err := e.setupTracing(tctx)
+		e.tracingExporterShutdown = func() { tracingExporter.Shutdown(tctx) }
```

Comment:
Shell we register it even if e.setupTracing() returned  (nil, err) ?

Reply:
Good catch, I added a check for either to be nil and I return a tracing setup error. Please have another look, thanks!